modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -928,5 +948,135 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
+    for (const auto & bside : side_list)
+    {
+      if (std::get<2>(bside) != boundary_id)
+        continue;
+
+      Elem * elem = mesh0->elem_ptr(std::get<0>(bside));
+      auto side = std::get<1>(bside);
+      std::unique_ptr<Elem> side_elem(elem->build_side_ptr(side));
+      auto copy = side_elem->build(side_elem->type());
+
+      Real l = side_elem->volume();
+      if (l < min_side)
+        min_side = l;
+
+      for (dof_id_type i : side_elem->node_index_range())
+      {
+        auto & n = side_elem->node_ref(i);
+
+        if (old_new_node_map.count(n.id()))
+          copy->set_node(i) = poly_mesh.node_ptr(old_new_node_map[n.id()]);
+        else
+        {
+          Node * node = poly_mesh.add_point(side_elem->point(i));
+          copy->set_node(i) = node;
+          old_new_node_map[n.id()] = node->id();
+        }
+      }
+      poly_mesh.add_elem(copy.release());
+    }
+    poly_mesh.skip_partitioning(true);
+    poly_mesh.prepare_for_use();
+
+    // triangulate with the 1d mesh
+    libMesh::Poly2TriTriangulator poly2tri(poly_mesh);
+    poly2tri.triangulation_type() = libMesh::TriangulatorInterface::PSLG;
+    poly2tri.set_interpolate_boundary_points(0);
+    poly2tri.set_refine_boundary_allowed(false);
+    poly2tri.set_verify_hole_boundaries(false);
+    const auto desired_area = getParam<Real>("inner_ring_desired_area");
+    const auto desired_area_func = getParam<std::string>("inner_ring_desired_area_func");
+    if (desired_area != 0)
+    {
+      poly2tri.desired_area() = desired_area;
+      if (desired_area_func != "")
+        paramError("inner_ring_desired_area_func",
+                   "cannot be specified with 'inner_ring_desired_area'");
+    }
+    else if (desired_area_func != "")
+    {
+      poly2tri.desired_area() = 0;
+      libMesh::ParsedFunction<Real> area_func{desired_area_func};
+      poly2tri.set_desired_area_function(&area_func);
+    }
+    else
+      poly2tri.desired_area() = min_side * min_side;
+    poly2tri.minimum_angle() = 0;
+    poly2tri.smooth_after_generating() = true;
+    poly2tri.elem_type() = libMesh::ElemType::TRI3;
+    // let us keep the center point
+    poly_mesh.add_point(libMesh::Point());
+    poly2tri.triangulate();
+    // keep the old subdomain id
+    for (auto elem : poly_mesh.element_ptr_range())
+      elem->subdomain_id() = _ring_block_ids[0];
+
+    if (isParamValid("ring_id_name"))
+    {
+      if (_ring_intervals[0] != 1 && getParam<MooseEnum>("ring_id_assign_type") == "ring_wise")
+        paramError("replace_inner_ring_with_delaunay_mesh",
+                   "Inner ring has multple intervals with each being assigned with a different "
+                   "ring id, replacing inner ring with Delaunay mesh will remove this ring id "
+                   "assign type. Either change 'ring_id_assign_type' to block_wise or set the "
+                   "first element of 'ring_intervals' to 1 or set "
```

Comment:
I think we should enforce the first element of `ring_intervals` to be 1 when this replacement is used.

Reply:
Yes, not enforcing this would make the input switch convenient. But it would also make the input syntax confusing for new users.

Maybe we can override the user provided input parameters when this replacement is used.

Anyway, it occurred to me that I forgot to mention that this new feature needs to be described in the documentation.